src/Microsoft.TestPlatform.Common/Filtering/FilterExpressionWrapper.cs

```diff
@@ -60,25 +113,28 @@ public string ParseError
         public string[] ValidForProperties(IEnumerable<String> supportedProperties, Func<string, TestProperty> propertyProvider)
         {
             string[] invalidProperties = null;
-            if (null != this.filterExpression)
+
+            if (UseFastFilter)
+            {
+                invalidProperties = this.fastFilter.ValidForProperties(supportedProperties);
+            }
+            else if (UseFilterExpression)
```

Comment:
Do we need a `UseFilterExpression`? Slightly confusing since it seems it is just a mutual exclusion to `UseFastFilter` although not all code paths ensure a null check to `filterexpression`. Maybe we can just make it a null check as before instead?

Reply:
I didn't set `ParseError` when there's regex parsing error, in which case, both `filterExpression` and `fastFilter` would be `null`. Seems `TestCaseFilterExpression.validForMatch` would short-circuit validation and evaluation if `ParseError` is set. Will fix.